kotlin/reflect/src/main/java/com/squareup/moshi/kotlin/reflect/KotlinJsonAdapter.kt

```diff
@@ -164,7 +183,25 @@ internal class KotlinJsonAdapter<T>(
   }
 }
 
-class KotlinJsonAdapterFactory : JsonAdapter.Factory {
+/**
+ * Failure strategy when a required value is missing or non-null value is null
+ */
+enum class FailureStrategy {
+
+    /**
+     * Throw a [JsonDataException] on the first violation
+     */
+    FAIL_FAST,
+
+    /**
+     * Accumulate all the violations and throw a [JsonDataException] at the end
+     */
+    ACCUMULATE_ERRORS
+}
+
+class KotlinJsonAdapterFactory @JvmOverloads constructor(
```

Comment:
applying this to all classes might be hard to replicate in code gen. I wonder if it'd be worth putting this in `@JsonClass` as a property instead and make them all opt-in.